mlflow/models/docker_utils.py

```diff
@@ -73,6 +73,7 @@ def generate_dockerfile(
     mlflow_home: Optional[str] = None,
     enable_mlserver: bool = False,
     disable_env_creation_at_runtime: bool = True,
+    install_java: bool = True,
```

Comment:
Let's set the default value to None.
Then in below:
1. if install_java is not None and base_image.startswith("python:") --> emit a warning message saying we're ignoring this option and use python image as base image
2. if base image is UBUNTU, we add the setup_java step if install_java is not False